include/revng/Model/Pass/RegisterModelPass.h

```diff
@@ -16,27 +17,50 @@ class RegisterModelPass {
   using ModelPass = std::function<void(TupleTree<model::Binary> &)>;
 
 private:
-  static llvm::ManagedStatic<std::map<std::string, ModelPass>> Registry;
+  struct Registered {
+    std::string Name;
+    std::string Description;
+    ModelPass Pass;
+  };
+  struct TransparentNameComparator {
+    using is_transparent = std::true_type;
+
+    template<typename Type>
+    llvm::StringRef get(const Type &Value) const {
+      return llvm::StringRef(Value);
+    }
+    llvm::StringRef get(const Registered &Value) const {
+      return llvm::StringRef(Value.Name);
+    }
+    template<typename LHS, typename RHS>
+    bool operator()(const LHS &Left, const RHS &Right) const {
+      return get(Left) < get(Right);
+    }
+  };
```

Comment:
What does this enable you to do? You can lookup stuff using a `StringRef` instead of a `std::string`?
Kinda cool.

Maybe we can make this more generic moving it to its own headers and letting it accept a lambda returning `Value.name`. This way we can make any `std::set`/`std::map` using a string-like object as a key querable using a string-like object (`Twine`, `SmallString`, `StringRef`, `std::string`).

Reply:
Well, yeah, sure, here I just habitually added a transparent comparation because there's not much reason not to - if you don't like it, feel free to just go back to a normal `std::less`.

Reply:
What I meant is that I'd be happy to see a generic version of this. But not today.

Reply:
Ah, sorry for misunderstanding. But seriously, don't we all? :D